CONTRIBUTING.md

```diff
@@ -2,31 +2,109 @@
+If you would like to request a feature that does not yet exist in KubeCF you should do a quick [search][2] - someone
+may already have opened the same feature request. If not, go ahead and open a new issue by selecting the "Feature" issue type and answer some needed questions.
 
 ## Code review process
 
-The core team looks to Pull Requests on a regular basis and in a best effort.
+The core team looks at Pull Requests on a regular basis on a best-effort basis.
+
+## How are Github issues handled
+
+### Columns
+
+1. Icebox: New issues that require further grooming. Note that some issues may be closed or rejected.
+
+2. To Do: issues ordered by priority and ready to be picked by contributors.
+
+3. In progress: issues being implemented.
+
+4. Under review: issues waiting to be reviewed.
+
+5. Done: all issues closed.
+
+### Labels
+
+For tracking purposes, issues are labeled based on their **priority**, **status**, **type** and **size**.
+There're some individual labels like **good first issues** or **help wanted**, that can be used but are not tracked from
+a project management perspecitve.
+
+All the grouping labels must contain a short description to help contributors to understand better the intention of
+each one.
+
+#### Status
+
+1. **Accepted**: issue was accepted and it may be implemented in near future.
+2. **Need More Info**: issue requires more information to be evaluated.
+3. **Verification Needed**: issue probably waiting for a PR to be merged.
+4. **Validation**: issue has enough information but it requires a discussion before accepted.
+
+[todo] more about the labels grouping in future versions of the document.
+
+### Creating New Issues
+
+The issue created on Github will be automatically assigned to the product owner, who will then evaluate it.
+
+If the issue does not contain sufficient information the author will be requested to provide additional details.
+
+When an issue goes "Stale" (more than 60 days without any activity), the product owner will:
+
+1. close the issue in case of lack of information.
+2. close the issue if it's no longer relevant.
+3. change labels if it's still relevant and information is sufficient.
+
+### Where to start
+
+All contributions are welcome but the best place to pick an issue to work on is from the _todo_ column that is ordered by
+priority. Before you move it to the _in progess_ column, make sure that the story is clear. If not, add your comments and/or 
+questions to the issue.
+
+### How do we estimate
+
+We use t-shirt sizes (XS, S, M, L, XL, XXL)  to determine the Github issue complexity and not the implementation effort, so we can determine if the issue needs to be split in multiple ones and/or if it requires a more extended brainstorm to determine the feasibility of the new feature.
```

Comment:
I don't see any relation with the aforementioned labels, are those going to be new ones or there is an implicit size relation?

Reply:
new ones